DietPi-Software | TasmoAdmin: Fixe firmware restriction error

SimonPHP opened the pull request:
Tasmoadmin could not deploy update to clients because the `/tasmoadmin/data/firmwares` directory had restricted access. Additionally, the path given in this file was wrong. `/tasmoadmin/firmwares` doesn't exist.

<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->

SimonPHP committed:
Fixed restriction error

Tasmoadmin could not deploy update to clients because the `/tasmoadmin/data/firmwares` directory had restricted access. Additionally, the path given in this file was wrong. `/tasmoadmin/firmwares` doesn't exist.

MichaIng commented on .conf/dps_27/nginx.tasmoadmin.conf:
```diff
@@ -1,6 +1,8 @@
+	# Allow access to firmwares
+	location ~ ^/tasmoadmin/data/firmwares {
+                allow all;
+        }
```
This should work as well and is more efficient: `^~` matches a prefix and then stops searching, i.e. no other location blocks are applied.
```suggestion
	location ^~ /tasmoadmin/data/firmwares {
        }
```

MichaIng committed:
v8.17

- DietPi-Software | TasmoAdmin: Grant access to firmwares for Lighttpd as well, align comments and make syntax slightly more efficient, using "^~" prefix+stop operator on Nginx for the firmwares path

MichaIng reviewed (approved):
Works fine for me on all webservers.
